web/concrete/src/Permission/Key/Key.php

```diff
@@ -502,4 +502,10 @@ public static function exportTranslations()
 
         return $translations;
     }
+
+    public static function clearCache()
+    {
+        $cache = Core::make('cache/expensive');
```

Comment:
I see what you mean, but I don't think this is part of this PR. Refactoring those classes is a different task to me and just refactoring the permission key class alone won't do much good..